src/types/kubectl.ts

```diff
@@ -147,8 +147,9 @@ export class Kubectl {
     if (this.ignoreSSLErrors) {
       args.push("--insecure-skip-tls-verify");
     }
-    args = args.concat(["--namespace", this.namespace]);
-
+    if (this.namespace != "default") {

```

Comment:
we should also check if this.namespace == ""

this.namespace != "default" && this.namespace != ""